src/Symfony/Component/Console/Application.php

```diff
@@ -181,14 +181,25 @@ public function run(InputInterface $input = null, OutputInterface $output = null
         return $exitCode;
     }
 
+    /**
+     * Checks if the current application has the parameter in the default option.
+     */
+    final protected function hasDefaultOption(string $option): bool
+    {
+        $defaultDefinition = $this->getDefaultInputDefinition();
+
+        return $defaultDefinition->hasOption($option);
+    }
+
     /**
      * Runs the current application.
      *
      * @return int 0 if everything went fine, or an error code
      */
     public function doRun(InputInterface $input, OutputInterface $output)
     {
-        if (true === $input->hasParameterOption(['--version', '-V'], true)) {
+        // If no command was used, and the default version option was not overriden, print current version and return.
+        if ($this->hasDefaultOption('version') && true === $input->hasParameterOption(['--version', '-V'], true)) {
```

Comment:
While editing these lines, I'd suggest removing the comparison to `true` on all of them.